harness/harness-core-ui
# feat: [CDS-86868] search and pagination for infra definitions

### Summary

- Added pagination support https://harness.atlassian.net/browse/CDS-86868
- Added search support https://harness.atlassian.net/browse/CDS-86867

#### Screenshots

- Pagination

https://github.com/harness/harness-core-ui/assets/94995410/a32a5338-a0cb-4cf9-8964-99a5d2d4fe6b

- Search 

https://github.com/harness/harness-core-ui/assets/94995410/62c70ed0-4d3c-4d5d-a3ef-a4818c404266

### Manual Testing
- created new env, new infra check for infra definitions listing 
- check for appropriate search result based on search term
- checked for page size and page navigation



#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)
